components/Canvas.tsx

```diff
@@ -16,43 +16,47 @@ export function useCanvas() {
 }
 
 export function useContext2d() {
-  const context2d = useContext(Context2DContext);
+  const context2d = useContext(Context2dContext);
   if (!context2d) {
     throw new Error("useContext must be used within a Canvas component");
   }
 
   return context2d;
 }
 
-export function Canvas({ children }: { children: preact.ComponentChildren }) {
-  const canvasRef = useRef<HTMLCanvasElement>(null);
-  const contextRef = useRef<CanvasRenderingContext2D | null>(null);
-  const [isMounted, setIsMounted] = useState(false);
+export type CanvasProps = { children?: preact.JSX.Element | null };
+
+export function Canvas({ children }: CanvasProps) {
+  const [canvasElement, setCanvasElement] = useState<HTMLCanvasElement | null>(
+    null
+  );
   const [width, setWidth] = useState(0);
   const [height, setHeight] = useState(0);
-  useLayoutEffect(() => {
+
+  useEffect(() => {
```

Comment:
only using the effect now to register the event, which is how it should of been done in the first place.